sdk/storage/storage-blob/src/Clients.ts

```diff
@@ -8100,8 +8100,7 @@ export class ContainerClient extends StorageClient {
    * @param {ContainerListBlobsOptions} [options={}] Options to list blobs operation.
    * @returns {(PagedAsyncIterableIterator<
    *   { kind: "prefix" } & BlobPrefix | { kind: "blob" } & BlobItem,
-   *     ContainerListBlobHierarchySegmentResponse
-   *   >)}
+   *     ContainerListBlobHierarchySegmentResponse>)}
```

Comment:
I don't know the requirements for writing these doc things.
Both the official API and the github.io one look weird for the returns now.
https://docs.microsoft.com/en-us/javascript/api/@azure/storage-blob/containerclient?view=azure-node-latest#listBlobsByHierarchy_string__ContainerListBlobsOptions_

https://azuresdkdocs.blob.core.windows.net/$web/javascript/azure-storage-blob/12.3.0/classes/containerclient.html#listblobsbyhierarchy

Maybe doc the `returns` with one line would be safer? Not able to verify it till we release it though.

Reply:
Do we even need to have the type defined in the `@returns` tag? I would assume all tooling would infer the return type from the method signature. The `@returns` tag can then be used only if we want to add some text describing the return value rather than defining the type.

Reply:
@ljian3377 We will use https://github.com/Azure/azure-sdk/issues/2084 to come up with requirements and update the guidelines